### DISCUSSION TITLE
Permanently loaded chunks

### DISCUSSION THREAD

**Discussion by ZakShearman** [Minestom/Minestom#846] [Help] [answered]
How can you load all chunks within an Instance and then prevent any chunks from unloading (using AnvilLoader)?

**Comment by RinesThaix** [comment]
From what I can see from sources, chunks don't unload unless you force them to do so or unregister the whole instance.
What about loading, just spam Instance#loadChunk for all the chunks you need.

**Reply by ZakShearman** [reply]
I wouldn't have knowledge of the specific chunk location that needs to be loaded and it would be extremely painful to have to specify that information in a static manner.

As for chunk unloading that does seem to be the case.

**Reply by RinesThaix** [reply]
Then you can use dynamic programming approach and at each step load a chunk, then try to load all of its neighbours. If chunk can't be loaded, Minestom will try to generate it, so just provide it with an empty generator: that should work.

**Comment by emortaldev** [comment, marked as answer]
Chunks are not unloaded by default

If you want to unload them when finished, you can use
```java
MinecraftServer.getGlobalEventHandler().addListener(PlayerChunkUnloadEvent.class, e -> {
    var chunk = e.getInstance().getChunk(e.getChunkX(), e.getChunkZ());
    if (chunk == null) return

    if (chunk.getViewers().size() == 0) {
        try {
            e.getInstance().unloadChunk(e.getChunkX(), e.getChunkZ());
        } catch (NullPointerException ignored) {}
    }
});
```

**Reply by ZakShearman** [reply]
Ahh didn't realise they weren't unloaded by default.

Any way to also load all chunks in an instance?

**Comment by MrlnHi** [comment]
Making your own ChunkLoader is probably the best way

**Reply by ZakShearman** [reply]
Do any of the other world loading libraries for minestom support loading all chunks that you know? I'd prefer not having to entirely rewrite the anvil loader.
